src/providers/TransactionsDispatcherContextProvider.tsx

```diff
@@ -484,76 +548,103 @@ const TransactionsDispatcherContextProvider = ({ children }: { children: ReactNo
   }, [crossChainActions, getSdkForChainId, getEtherspotPrimeSdkForChainId]);
 
   useEffect(() => {
+    let timeout: any;
     let updateExpired = false;
 
     const validPendingWeb3CrossChainActions = crossChainActions.filter(
-      (crossChainAction) =>
-        crossChainAction.useWeb3Provider &&
+      (crossChainAction) => crossChainAction.useWeb3Provider &&
         getCrossChainActionTransactionsByStatus(crossChainAction.transactions, CROSS_CHAIN_ACTION_STATUS.PENDING).length
     );
 
-    const checkWeb3TransactionStatuses = () => {
-      validPendingWeb3CrossChainActions.map(async (crossChainAction) => {
+    const checkWeb3TransactionStatuses = async () => {
+      await Promise.all(validPendingWeb3CrossChainActions.map(async (crossChainAction) => {
         const sdkForChain = isEtherspotPrime(etherspotMode)
           ? await getEtherspotPrimeSdkForChainId(crossChainAction.chainId)
           : getSdkForChainId(crossChainAction.chainId);
+
         if (!sdkForChain) return;
 
-        await Promise.all(
-          crossChainAction.transactions.map(async (transaction) => {
-            if (!transaction?.transactionHash || updateExpired) return;
+        await Promise.all(crossChainAction.transactions.map(async (transaction) => {
+          if (!transaction?.transactionHash || updateExpired) return;
 
+          let status: string | undefined;
+
+          if (transaction?.status === CROSS_CHAIN_ACTION_STATUS.PENDING
+            && crossChainAction.type === TRANSACTION_BLOCK_TYPE.KLIMA_STAKE) {
+            try {
+              const crossChainStatus = await getCrossChainStatusByHash(
+                crossChainAction.chainId,
+                crossChainAction.destinationCrossChainAction[0].chainId,
+                transaction.transactionHash,
+              );
+
+              if (crossChainStatus?.status
+                && crossChainStatus?.subStatus
+                && crossChainStatus.status.toLowerCase() === 'done'
+                && crossChainStatus.subStatus.toLowerCase() === 'completed') {
+                status = CROSS_CHAIN_ACTION_STATUS.CONFIRMED;
+              }
+            } catch (e) {
+              //
```

Comment:
not really, unless we want to wrap any error that comes from endpoint, it always starts as 500 bad request because the tx hash is not yet mined and whatever rpc service is relying to it's better just try try hitting it until completes, other solution would be to try break down other kind of errors, but considering time constraints and rest of the code I don't think it makes sense now, wdyt?

Reply:
Ahh okay i didn't realise the behaviour was like that - in that case this would not have the intended effect